x-pack/test/cases_api_integration/security_and_spaces/tests/trial/internal/get_user_action_stats.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import http from 'http';
+import expect from '@kbn/expect';
+
+import { ConnectorTypes } from '@kbn/cases-plugin/common/api';
+import { ObjectRemover as ActionsRemover } from '../../../../../alerting_api_integration/common/lib';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import { createCase, deleteAllCaseItems, pushCase, updateCase } from '../../../../common/lib/utils';
+import { postCaseReq } from '../../../../common/lib/mock';
+import {
+  createCaseWithConnector,
+  createConnector,
+  getJiraConnector,
+  getServiceNowSimulationServer,
+} from '../../../../common/lib/connectors';
+import { getCaseUserActionStats } from '../../../../common/lib/user_actions';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const actionsRemover = new ActionsRemover(supertest);
+
+  describe('get_case_user_action_stats', () => {
+    let serviceNowSimulatorURL: string = '';
+    let serviceNowServer: http.Server;
+
+    before(async () => {
+      const { server, url } = await getServiceNowSimulationServer();
+      serviceNowServer = server;
+      serviceNowSimulatorURL = url;
+    });
+
+    afterEach(async () => {
+      await deleteAllCaseItems(es);
+      await actionsRemover.removeAll();
+    });
+
+    after(async () => {
+      serviceNowServer.close();
```

Comment:
💯

Reply:
This was the "closing ports/sockets we opened in tests" you mentioned right? ;)

Reply:
Yep haha.